ui/src/helpers/calc-overall-vulnerability.ts

```diff
@@ -29,7 +29,7 @@ import { VulnerabilityRating } from '@/helpers/get-status-class';
 // Now that we have scores and vectors available, we could do better with
 // calculation of the overall vulnerability rating.
 export function calcOverallVulnerability(
-  ratings: (number | undefined)[]
+  ratings: (number | null | undefined)[]
```

Comment:
The `null` should be taken into account in the below filter. The `rating is null` means that, as TypeScript wasn't able to narrow the type on it's own, we told is that if we filter undefineds away, all of the ratings are numbers. After adding nulls as another option, this is no longer the case.

Luckily TypeScript recent release made this better, so the whole `is number` can be removed. The filter should be changed to:

```typescript
  const filteredRatings = ratings.filter(
    (rating) => rating !== undefined && rating !== null
  );
```